### ISSUE TITLE
If this issue gets at least one comment. I'll write a documentation

### ISSUE THREAD

**Issue by Citillara** [Citillara/IrcClient#4]

**Comment by AuracleTech** [comment]
Comment.jpeg

**Comment by AuracleTech** [comment]
For real tho I'd needs docs, trying to make an app to connect to osu IRC system.
If you ever update and make docs can you also make a quick nuget package 💖

**Comment by Citillara** [comment]
Damn guys !!! 4 Years after. Ok I'll start writing it

**Comment by Citillara** [comment]
Yes this one has been tested againt osu! IRC, which has a kinda strange implementation. It works on it well

**Comment by AuracleTech** [comment]
Yeah I'm using it for osu, if you could update it from .Net Framwork to .Net Core it would be epic
Cause I might have to stop using it due to the compatibility issue

**Comment by AuracleTech** [comment]
As osu libs are in .Net Core

**Comment by Citillara** [comment]
Sure, I'll look into it. I was looking for an excuse to go poke my nose in .Net Core since a while

**Comment by Citillara** [comment]
Ported to .NetCore

**Comment by Valinwolf** [comment]
Latest commit is less than half a year old, so I assume you are still actively developing this? If so how's the docs & Nuget package coming? I'm developing my own private encrypted chat system and wanted to use IRC as the backbone of the system. Looking for a cross-platform C# library with a decent set of docs and high ease of use has been a struggle lol.

**Comment by Citillara** [comment]
Still actively developped and maintained. I started writing the doc locally, didn't pushed anything yet sorry. I was considering a small refractor on certain types (especially NOTICE from services) as helper

**Comment by Citillara** [comment]
I've been using this client since 2014 on both Windows (.NetFramwork) and Linux (Mono) without needing any major changes/bugs. I'm still using it as com protocol for other projects. 
Features tends to implemented on my need basis

**Comment by Valinwolf** [comment]
> Still actively developped and maintained. I started writing the doc locally, didn't pushed anything yet sorry. I was considering a small refractor on certain types (especially NOTICE from services) as helper

Sweet! I look forward to seeing it!

> I've been using this client since 2014 on both Windows (.NetFramwork) and Linux (Mono) without needing any major changes/bugs. I'm still using it as com protocol for other projects. 
> Features tends to implemented on my need basis

I'm using it for a Win/Linux/Android project. I've been just experimenting with the library and looking over the sample code to fill in the gaps of what I couldn't figure out. Win/Linux seems to work great, but when I tried to fire up the Android project, connection failed. Over all seems to be the most likely candidate for my project.

**Comment by Citillara** [comment]
Meanwhile you could look at my other projects that are using it

MemorySoulLink
https://github.com/Citillara/MemorySoulLink/blob/main/Program.cs#L85

CitiBot
https://github.com/Citillara/CitiBot
which uses my TwitchClient
https://github.com/Citillara/TwitchClient

TwitchClient is a wrapper of the IrcClient for CitiBot that handles everything specific to TwitchServer (the odd way to send a private message to a specifc channel to send a whisper, or simply all the information store in the IRCv3 tags (usernames, ids...)

CitiBot is a framework made for connecting multiple bots to TwitchChat with a database integration
